### ISSUE TITLE
Clash between Qonversion SDK and In_app_purchase package

### ISSUE THREAD

**Issue by gmiszewski-intent** [qonversion/flutter-sdk#137]
Hello,

I have a problem integrating QonversionSdk for Android in observeMode.
`In_app_purchase` Flutter library uses newer version of native BillingClient than this library. 

---------------

Latest Flutter version of in_app_purchase uses native library which uses BillingClient `3.0.2`
https://github.com/flutter/plugins/blob/main/packages/in_app_purchase/in_app_purchase_android/android/build.gradle

Tagged version of latest quonversion for flutter uses Android QonversionSdk for android in version `3.2.4`:
https://github.com/qonversion/flutter-sdk/blob/4.3.1/android/build.gradle

Lastly Android QonversionSdk for unknown reason uses BillingClient `2.2.0`
https://github.com/qonversion/android-sdk/blob/3.2.4/app/build.gradle 

I've checked issues in android-sdk and they say it should be using totally different version of BillingClient
https://github.com/qonversion/android-sdk/issues/288#issuecomment-1014812019

--------------

I don't know why `in_app_purchase` doesn't use 4.0 version, but for me this library should be compatible with official purchase library for Flutter.

**Comment by SpertsyanKM** [comment]
Hello, @gmiszewski-intent! Thanks for contacting us.

As I see you want to implement our SDK which supports Google Billing library version 3. 

The version of Flatter SDK you provided uses Android-SDK which depends on Google Billing 4+. The [file](https://github.com/qonversion/android-sdk/blob/3.2.4/app/build.gradle) you provided is the app-level build.gradle (it is our sample app), not the [SDK-level](https://github.com/qonversion/android-sdk/blob/3.2.4/sdk/build.gradle), which you are interested in. As you can see, here we use Google Billing 4.0.0.

So what you need is to use Flutter SDK version 3+ (3.5.0 is the latest now). It [uses](https://github.com/qonversion/flutter-sdk/blob/3.5.0/android/build.gradle) our Android-SDK version 2.11.7, which in turn [uses](https://github.com/qonversion/android-sdk/blob/2.11.7/sdk/build.gradle) Google Billing 3.0.0.

So, long story short - 
Flatter SDK version 3 is still supported for those who use Google Billing 3. For Google Billing 4 you need to use Flatter SDK version 4.

**Comment by SpertsyanKM** [comment]
Glad to help!
